features/steps/ios_steps.rb

```diff
@@ -87,25 +87,59 @@
   end
 end
 
-Then("each event in the payload matches one of:") do |table|
-  # Checks string equality of event fields against values
-  events = read_key_path(Server.current_request[:body], "events")
-  table.hashes.each do |values|
-    assert_not_nil(events.detect do |event|
-      values.all? do |k, v|
-        if k.start_with? 'has '
-          event_value = read_key_path(event, k.split(' ').last)
-          if v == 'yes'
-            !event_value.nil?
-          else
-            event_value.nil?
-          end
-        else
-          v == read_key_path(event, k) || (v.to_i == read_key_path(event, k).to_i)
-        end
-      end
-    end, "No event matches the following values: #{values}")
+Then("the received requests match:") do |table|
+  # Checks that each request matches one of the event fields
+  requests = Server.stored_requests
+  request_count = requests.count()
+  match_count = 0
+
+  # iterate through each row in the table. exactly 1 request should match each row.
+  table.hashes.each do |row|
+    requests.each do |request|
+      event = read_key_path(request[:body], "events.0")
+      match_count += 1 if request_matches_row(event, row)
+    end
   end
+  assert_equal(request_count, match_count, "Unexpected number of requests matched the received payloads")
+end
+
+def request_matches_row(body, row)
+  request_matches = true
+
+  row.all? do |key, expected_value|
+    obs_val = read_key_path(body, key)
+    match = false
+    if obs_val.nil?
```

Comment:
I don't think this is right - expected will always match if observed is nil, even if it isn't itself.  I'm looking at the tests myself as well now though, so can look at handling this.